### PR TITLE
Reintroducing Spring Cloud Services Config Client in apps.

### LINKED ISSUE #71: Reintroduce  Spring Cloud Services Config Client

### PR DESCRIPTION
**Pull request by sobychacko**
- Enabled spring-cloud-services-starter-config-client in app starters core
 - Fixed tests in CloudFoundryMicrometerCommonTagsTest by mocking the vcap configuration
 - Reused some utility classes/code from the java-cfenv project

Fixes #71

### CODE REVIEW

File: common/app-starters-micrometer-common/src/test/java/org/springframework/cloud/stream/app/micrometer/common/AbstractMicrometerTagTest.java [hunk 1]
```diff
@@ -58,6 +67,85 @@ public void before() {
...
+		try {
+			Map<String, Object> serviceMap = objectMapper.readValue(servicePayload,
+					Map.class);
+			return serviceMap.get("label").toString();
+		}
+		catch (Exception e) {
+			return null;
+		}
+	}
+
```
Comment (tzolov) [inline common/app-starters-micrometer-common/src/test/java/org/springframework/cloud/stream/app/micrometer/common/AbstractMicrometerTagTest.java]: We don't need this  json re-writing code. Instead we can change our test json to something like this:
```
{
  "sso":[{
    "name": "sso",
    "label": "sso",
    "plan": "notfree",
    "tags": ["configuration"],
    "credentials":{
      "uri": "http://pivotal.io",
      "client_id": "fakeClientId",
      "client_secret": "fakeSecret",
      "access_token_uri": "token"
    }
  }]}
``` 
And then add it to the env variables like this:
```
	@BeforeClass
	public static void setup() throws IOException {
		String serviceJson = StreamUtils.copyToString(new DefaultResourceLoader().getResource(
				"classpath:/org/springframework/cloud/stream/app/micrometer/common/pcf-scs-info.json")
				.getInputStream(), Charset.forName("UTF-8"));
		CfEnvTestUtils.mockVcapServicesFromString(serviceJson);
	}
```
No extra code is required

File: common/app-starters-micrometer-common/src/test/resources/org/springframework/cloud/stream/app/micrometer/common/cloud-test-info.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "label": "sso",
+  "plan": "notfree",
+  "tags": ["configuration"],
+  "credentials":{
+    "uri": "http://pivotal.io",
+    "client_id": "fakeClientId",
+    "client_secret": "fakeSecret",
+    "access_token_uri": "token"
+  }
+}
```
Comment (tzolov) [inline common/app-starters-micrometer-common/src/test/resources/org/springframework/cloud/stream/app/micrometer/common/cloud-test-info.json]: See my comment above for the json content.

Comment (tzolov) [reply common/app-starters-micrometer-common/src/test/resources/org/springframework/cloud/stream/app/micrometer/common/cloud-test-info.json]: @sabbyanandan i've submitted a PR on top of your PR ;)(https://github.com/sobychacko/core/pull/1) that applies the above-suggested changes. 

Or if you are fine with those changes i can apply them and merge it my self?

Comment (sobychacko) [reply common/app-starters-micrometer-common/src/test/resources/org/springframework/cloud/stream/app/micrometer/common/cloud-test-info.json]: @tzolov Please go ahead with your changes and suggestions and merge. (The only reason I put that new artifact as `test` scope in the BOM is that it is already a test artifact and doesn't contain any non-test code, but please go with what you said for consistency reasons)

**Comment by tzolov** [comment]
Updated and merged at ca5fab8f2eb65dc9a51fd76a4da9ba76f4b5c190

**Pull request closed without merging**